frontend/components/EnrollSecretTable/EnrollSecretTable.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+
+import { IEnrollSecret } from "interfaces/enroll_secret";
+import EnrollSecretRow from "./EnrollSecretRow";
+
+const baseClass = "enroll-secrets";
+
+interface IEnrollSecretRowProps {
+  secrets: IEnrollSecret[] | undefined;
```

Comment:
I wonder if this prop ever needs to be `undefined`? If it can come in as a required prop you could remove the checks below.

Reply:
You can delete all your secrets, so I think that's where it can be undefined?

Reply:
#2933